language/tools/move-package/src/resolution/resolution_graph.rs

```diff
@@ -515,6 +515,33 @@ impl ResolvingGraph {
         }
     }
 
+    pub fn download_dependency_repos(
+        package: SourceManifest,
+        build_options: BuildConfig,
+        root_path: PathBuf,
+    ) -> Result<()> {
+        // include dev dependencies if in dev mode
+        let additional_deps = if build_options.dev_mode {
+            package.dev_dependencies.clone()
+        } else {
+            BTreeMap::new()
+        };
+
+        for (dep_name, dep) in package
+            .dependencies
+            .into_iter()
```

Comment:
should this just be an `iter()`? Then everything else here can be by reference?

Reply:
I've updated the code, hopefully it's what you had in mind.

Reply:
Yes, but in 
```
        if config.fetch_deps_only {
            config.download_deps_for_package(&rerooted_path)?;
            return Ok(());
        }
```
https://github.com/move-language/move/pull/267/files#diff-5d804e276a879688a14ffe2fc7342accb0050d6e84418a8cd855a9faa93d1ce4R18
`config` is destroyed after the `download_deps_for_package` anyway, so it doesn't need to be a reference. 

Same for the `SourceManifest`, it's not used outside the `download_dependency_repos` call, so there's no need for a reference. Unless I'm missing something. 

I've fixed the `root_path`, it does remove one `clone()`.

Reply:
> config is destroyed after the download_deps_for_package anyway

Sure, but you aren't actually using it by-value in any meaningful or required way. So if anyone needs to modify surrounding code afterwards, they are now stuck.

It is best to use things by-reference unless you have some reason not to (prevents clones, ensures proper mutations, etc)

Reply:
Could you review it again? I made everything into a by-reference, and extracted the only mutation required for the `mut self` to the higher scope.